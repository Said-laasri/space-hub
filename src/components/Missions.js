import { useSelector, useDispatch } from 'react-redux';
import { useEffect } from 'react';
import Table from 'react-bootstrap/Table';
import { getMission } from '../redux/missions/missions';
import MissionOne from './missionOne';

let isInitial = true;

function Missions() {
  const dispatch = useDispatch();
  const missions = useSelector((state) => state.mission);

  useEffect(() => {
    if (isInitial) {
      dispatch(getMission());
      isInitial = false;
    }
  }, [dispatch]);

  return (
    <div className="table-responsive">
      <Table striped bordered hover>
        <div className="theader">
          <div className="ul">
            <p className="mi">Mission</p>
            <p className="de">Description</p>
            <p className="st">Status</p>
          </div>
        </div>
        {missions.map((mission) => (
          <MissionOne
            key={mission.id}
            id={mission.id}
            name={mission.name}
            description={mission.description}
            reserved={mission.reserved}
          />
        ))}
      </Table>
    </div>
  );
}

export default Missions;
