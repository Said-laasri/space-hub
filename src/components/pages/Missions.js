import { useSelector, useDispatch } from 'react-redux';
import { useEffect } from 'react';
import Table from 'react-bootstrap/Table';
import { getMission } from '../redux/missions';
import MissionOne from '../missionOne';

function Missions() {
  const dispatch = useDispatch();
  const missions = useSelector((state) => state.mission);

  useEffect(() => {
    dispatch(getMission());
  }, [dispatch]);

  return (
    <Table striped bordered hover size="lg">
      <thead>
        <tr>
          <th>Mission</th>
          <th>Description</th>
          <th>Status</th>
        </tr>
      </thead>
      {missions.map((mission) => (
        <MissionOne
          key={mission.id}
          id={mission.id}
          name={mission.name}
          description={mission.description}
          reserved={mission.reserved}
        />
      ))}
    </Table>
  );
}

export default Missions;
