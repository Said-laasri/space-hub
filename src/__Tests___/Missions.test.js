import React from 'react';
import Renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configueStore';
import Missions from '../components/Missions';

it('renders correctly', () => {
  const tree = Renderer.create(
    <React.StrictMode>
      <Provider store={store}>
        <BrowserRouter>
          <Missions />
        </BrowserRouter>
      </Provider>
    </React.StrictMode>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
