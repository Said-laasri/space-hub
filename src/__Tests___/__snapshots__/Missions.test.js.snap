// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<table
  className="table table-lg table-striped table-bordered table-hover"
>
  <thead>
    <tr>
      <th>
        Mission
      </th>
      <th>
        Description
      </th>
      <th>
        Status
      </th>
    </tr>
  </thead>
</table>
`;
